### PR TITLE
NAS-117233 / 22.12 / NAS-117233: New Google Cloud Storage task field: `bucket_policy_only`

### PR DESCRIPTION
**Pull request by bvasilenko**
Testing:

At the **Data Protection > Cloud Sync Tasks > Add** (And also at **... > Update**)

There's a new checkbox field:

- **Title:** Bucket Policy Only

- **Description:** Access checks should use bucket-level IAM policies.

It should include a new item `bucket_policy_only` Into the `attributes` object at these server queries.

Back-end PR: https://github.com/truenas/middleware/pull/9421

### CODE REVIEW

**Review by undsoft — CHANGES_REQUESTED** [review]
I think this is only supposed to show for providers of type Google Cloud Storage.

File: src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts [hunk 1]
```diff
@@ -865,6 +877,12 @@ export class CloudsyncFormComponent implements FormConfiguration {
               this.setDisabled('bucket_input', true, true);
             }
 
+            if (targetProvider && targetProvider.name === 'GOOGLE_CLOUD_STORAGE') {
```
Comment (AlexKarpov98) [inline src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:880]: It'll better if the Provider name is used as an enum type, I think.

**Review by AlexKarpov98 — CHANGES_REQUESTED** [review]

File: src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts (hunk 1 above)
Comment (bvasilenko) [reply src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:880]: Ok, there's no suitable enum, so I'll ping you once I'll replace all the providers from this dropdown with the enum

image

**Review by AlexKarpov98 — APPROVED** [review]
The code change looks good!

**Pull request merged**